code/game/objects/items/weapons/tanks/tanks.dm

```diff
@@ -1,50 +1,41 @@
-/datum/action/item_action/tank/internals
-	name = "Set Internals"
-
-/datum/action/item_action/tank/internals/Trigger()
-	if(!Checks())
-		return
-
-	var/mob/living/carbon/human/C = owner
-	if(!istype(C))
-		return
-
-	if(C.internal == target)
-		C.internal = null
-		C << "<span class='notice'>You close \the [target] valve.</span>"
-		C.update_internals_hud_icon(0)
-	else if(C.wear_mask && (C.wear_mask.flags & MASKINTERNALS))
-		C.internal = target
-		C << "<span class='notice'>You open \the [target] valve.</span>"
-		C.update_internals_hud_icon(1)
-	return 1
-
-/datum/action/item_action/tank/internals/IsAvailable()
-	var/mob/living/carbon/C = owner
-	if(!C.wear_mask || !(C.wear_mask.flags & MASKINTERNALS))
-		return
-	return ..()
-
 /obj/item/weapon/tank
 	name = "tank"
 	icon = 'icons/obj/tank.dmi'
 	flags = CONDUCT
 	slot_flags = SLOT_BACK
 	hitsound = 'sound/weapons/smash.ogg'
-
 	pressure_resistance = ONE_ATMOSPHERE * 5
-
 	force = 5
 	throwforce = 10
 	throw_speed = 1
 	throw_range = 4
-
+	action_button_name = "Set Internals"
```

Comment:
the tank's action to set internals now use item/var/action instead of creating a new variable.